tests/test_special_ops.py

```diff
@@ -254,3 +256,40 @@ def test_accuracy_resolve_conj(shape, dtype):
     with flag_gems.use_gems():
         z = y.resolve_conj()
     assert not z.is_conj()
+
+
+@pytest.mark.parametrize("shape", UT_SHAPES_1D + UT_SHAPES_2D)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("n_samples", [1000])
+def test_accuracy_multinomial_with_replacement(shape, dtype, n_samples):
+    if shape[-1] == 1:
+        dist = torch.rand(size=shape, dtype=dtype, device="cuda")
+        with flag_gems.use_gems():
+            res_out = torch.multinomial(dist, n_samples, True)
+        assert torch.all(res_out == 0)
+    else:
+        # Mask p% off of the categories and test the sampling results fall in the rest
+        for p in (0.1, 0.5, 0.9):
+            dist = torch.rand(size=shape, dtype=dtype, device="cuda")
+            dist[torch.rand(shape) < p] = 0
+            # Make sure there's at least one non-zero probability
+            dist[..., -1] = 0.5
+            with flag_gems.use_gems():
+                res_out = torch.multinomial(dist, n_samples, True)
+            res_dist = torch.gather(dist, -1, res_out)
+            assert torch.all(res_dist)
+
+
+@pytest.mark.parametrize("pool", UT_SHAPES_1D + UT_SHAPES_2D)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_accuracy_multinomial_without_replacement(pool, dtype):
+    n_draws = 10
+    dist = torch.rand(size=(pool,), dtype=dtype, device="cuda").broadcast_to(
+        n_draws, pool
+    )
+    n_samples = pool
+    with flag_gems.use_gems():
+        res_out = torch.multinomial(dist, n_samples, False)
+    # Verifies uniqueness
+    sorted_samples, _ = res_out.sort(dim=1)
```

Comment:
for 1d shapes, I think the sort dim should be -1.

Reply:
In this test, the input dist and output samples are in 2d.

Reply:
but dim=-1 is better indeed.